text/off-chain-workers/indexing.md

```diff
@@ -0,0 +1,97 @@
+# Off-chain Indexing
+
+There are times when on-chain extrinsics need to pass data to the off-chain worker context with
+predictable write behavior. We can surely pass this piece of data via on-chain storage, but this is
+costly and it will make the data propagate among the blockchain network. If this is not a piece of
+information that need to have consensus upon the whole network, another way is to save this data
+in off-chain local storage via off-chain indexing.
+
+As off-chain indexing is called in on-chain context, **if it is agreed upon by the blockchain
+consensus mechanism, then it is expected to run predictably by all nodes in the network**. One use case
+is to store only the hash of certain information in on-chain storage for verification purpose but
+keeping the full data set off-chain for lookup later. In this case the original data can be saved
+via off-chain indexing.
+
+Notice as off-chain indexing is called and data is saved on block import, the result may be
+overridden should the block IS NOT FINALIZED eventually, or a fork appears in the blockchain.
```

Comment:
```suggestion
Notice as off-chain indexing is called and data is saved on every block import, this also includes forks.
The consequence of that is that in case non-unique keys are used the data might be overwritten by different
forked blocks and the content of off-chain database will be different between nodes.
Care should be taken when choosing the right indexing `key`, to prevent potential overwrites if it's not desired.
```

Reply:
updated in the latest commit